src/datajson.c

```diff
@@ -259,6 +259,36 @@ static int DatajsonAdd(
     return -1;
 }
 
+static uint32_t DatajsonAddStringElement(Dataset *set, json_t *value, char *json_key, bool *found)
+{
+    json_t *key = GetSubObjectByKey(value, json_key);
+    if (key == NULL) {
+        /* ignore error as it can be a working mode where some entries
+           are not in the same format */
+        return 0;
+    }
+
+    *found = true;
+
+    const char *val = json_string_value(key);
+    DataJsonType elt = { .value = NULL, .len = 0 };
+    elt.value = json_dumps(value, JSON_COMPACT);
+    elt.len = strlen(elt.value);
+
+    int add_ret = DatajsonAdd(set, (const uint8_t *)val, strlen(val), &elt);
+    if (add_ret < 0) {
+        FatalErrorOnInit("datajson data add failed %s/%s", set->name, set->load);
+        return 0;
```

Comment:
nit: `return` after `FatalError..` is a no-op

Reply:
This construct is used a lot and looking at the FatalErrorOnInit macro code, it seems it does not return (but exit if at init).

Reply:
Indeed. I mixed it up with the actual `FatalError` in my head. 🥲  Sorry for the noise.